src/main/java/ua/edu/internship/interview/web/controller/InterviewController.java

```diff
@@ -0,0 +1,91 @@
+import org.springframework.web.bind.annotation.RestController;
+import ua.edu.internship.interview.data.enumeration.InterviewStatus;
+import ua.edu.internship.interview.service.business.InterviewService;
+import ua.edu.internship.interview.service.dto.interview.InterviewCreateDto;
+import ua.edu.internship.interview.service.dto.interview.InterviewDto;
+import ua.edu.internship.interview.service.dto.interview.InterviewUpdateDto;
+import ua.edu.internship.interview.service.dto.interview.question.InterviewQuestionCreateDto;
+import ua.edu.internship.interview.service.dto.interview.question.InterviewQuestionDto;
+import ua.edu.internship.interview.service.dto.interview.question.InterviewQuestionUpdateDto;
+import java.util.List;
+
+@Tag(name = "Interviews", description = "Interviews management endpoints")
+@RestController
+@RequestMapping("/interviews")
+@RequiredArgsConstructor
+public class InterviewController {
+    private final InterviewService service;
+
+    @GetMapping
+    public List<InterviewDto> getAllInterviews() {
+        return service.getAllInterviews();
+    }
+
+    @GetMapping("/{interviewId}")
+    public InterviewDto getInterviewById(@PathVariable String interviewId) {
+        return service.getInterviewById(interviewId);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public InterviewDto createInterview(@RequestBody @Valid InterviewCreateDto dto) {
+        return service.createInterview(dto);
+    }
+
+    @PutMapping("/{interviewId}")
+    public InterviewDto updateInterview(@PathVariable String interviewId, @RequestBody @Valid InterviewUpdateDto dto) {
+        return service.updateInterview(interviewId, dto);
+    }
+
+    @DeleteMapping("/{interviewId}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void cancelInterview(@PathVariable String interviewId) {
+        service.deleteInterviewById(interviewId);
+    }
+
+
+//TODO Add comment in pull request, so it would be reasonable to add this endpoint
+
+    @PatchMapping("/{interviewId}/status")
+    public InterviewDto updateInterviewStatus(@PathVariable String interviewId, @RequestBody InterviewStatus status) {
+        return service.updateInterviewStatus(interviewId, status);
+    }
+
+    @PatchMapping("/{interviewId}/feedback")
+    public InterviewDto updateInterviewFeedback(@PathVariable String interviewId, @RequestBody String feedback) {
+        return service.updateInterviewFeedback(interviewId, feedback);
+    }
+
+    @PostMapping("/{interviewId}/questions")
+    @ResponseStatus(HttpStatus.CREATED)
+    public InterviewQuestionDto createInterviewQuestion(@PathVariable String interviewId, @RequestBody @Valid InterviewQuestionCreateDto dto) {
+        return service.createInterviewQuestion(interviewId, dto);
+    }
+
+    @PatchMapping("/{interviewId}/questions/{questionId}")
+    public InterviewQuestionDto updateInterviewQuestion(@PathVariable String interviewId, @PathVariable String questionId,
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.sizes.LineLengthCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Line is longer than 120 characters (found 122).
<!-- __reviewdog__:ChA0NmQ2ZTJlMDBlMDMwZmFlEgpjaGVja3N0eWxl -->